### PR TITLE
322 bacterias

### PR DESCRIPTION
**Pull request by alexander-myltsev**

### CODE REVIEW

File: parser/src/main/scala/org/globalnames/parser/FactoryAST.scala [hunk 1]
```diff
@@ -24,13 +34,19 @@ object FactoryAST {
...
-           genusParsed: Boolean = false): NodeMeta[Name] = {
+           genusParsed: Boolean = false)(implicit input: ParserInput): NodeMeta[Name] = {
+    val genus = input.sliceString(uninomial.node.pos.start, uninomial.node.pos.end)
+    val bacteria = bacteriaGenera.contains(genus)
     val name = Name(uninomial.node, subgenus.map { _.node }, species.map { _.node },
                     infraspecies.map { _.node }, comparison.map { _.node },
-                    approximation.map { _.node }, ignored, genusParsed)
+                    approximation.map { _.node }, ignored, bacteria, genusParsed)
+    val bacteriaHomonymWarning = bacteriaHomonymsGenera.contains(genus).option {
+      Vector(Warning(2, "Genus is bacteria name homonym", name))
```
Comment (alexander-myltsev) [inline parser/src/main/scala/org/globalnames/parser/FactoryAST.scala]: @dimus, what warning level and message should be?

File: parser/src/test/resources/test_data.txt [hunk 2]
```diff
@@ -1345,7 +1352,7 @@ Acastoides spp.
 9853f0a4-6324-5a7d-8108-e910578e612b|Acastoides spp.|Acastoides|Acastoides|||3
 
 Sphingomonas sp. 37
-{"name_string_id":"1daffd3a-f4de-58d9-91e3-ae4d08a50ce0","parsed":true,"quality":3,"quality_warnings":[[3,"Name is approximate"]],"parser_version":"test_version","verbatim":"Sphingomonas sp. 37","normalized":"Sphingomonas","canonical_name":{"value":"Sphingomonas"},"hybrid":false,"surrogate":true,"virus":false,"details":[{"genus":{"value":"Sphingomonas"},"annotation_identification":"sp.","ignored":{"value":"37"}}],"positions":[["genus",0,12],["annotation_identification",13,16]]}
+{"name_string_id":"1daffd3a-f4de-58d9-91e3-ae4d08a50ce0","parsed":true,"quality":3,"quality_warnings":[[3,"Name is approximate"]],"parser_version":"test_version","verbatim":"Sphingomonas sp. 37","normalized":"Sphingomonas","canonical_name":{"value":"Sphingomonas"},"hybrid":false,"surrogate":true,"virus":false,"details":[{"genus":{"value":"Sphingomonas"},"annotation_identification":"sp.","bacteria":true,"ignored":{"value":"37"}}],"positions":[["genus",0,12],["annotation_identification",13,16]]}
```
Comment (dimus) [inline parser/src/test/resources/test_data.txt]: ``bacteria:true`` should be at the same level as virus and surrogate

File: parser/src/main/scala/org/globalnames/parser/FactoryAST.scala (hunk 1 above)
Comment (dimus) [inline parser/src/main/scala/org/globalnames/parser/FactoryAST.scala]: this warning should not decrease quality, so the level would be 0. The value could be `The genus is a homonym of a bacterial genus`

**Comment by dimus** [comment]
LGTM, however we do have inconsistent behavior now, as surrogate and virus are always shown, but bacteria is not. I will merge as is, but we do need to undress the inconsistency

**Comment by alexander-myltsev** [comment]
Consistency brought to `bacteria` in PR: https://github.com/GlobalNamesArchitecture/gnparser/pull/352

Please, also resolve TODO: https://github.com/GlobalNamesArchitecture/gnparser/pull/351/files#diff-1fc4173837ddb1d2d68400e67e049c17R26

**Pull request merged**